ceph/rados/core_workflows.py

```diff
@@ -1980,3 +1971,36 @@ def restart_daemon_services(self, daemon: str):
 
         log.info(f"Ceph Orch Service(s) {daemon_services} has been restarted")
         return True
+
+    def check_pool_pg_states(self, pool: str, disallowed_states: list):
+        """
+        the method fetches the states for the pg belonging to particular pool.
+        If any of the PG in the pool, have states that are in the disallowed states list, returns fail
+
+        Args::
+            pool: Name of the pool whose pg states need to be checked
+            disallowed_states: list of pg states, that are not allowed, if found, module returns fail
+
+        Returns::
+            Pass -> True
+            Fail -> False
+        """
+        # fetching the pool ID
+        cmd = "ceph df"
+        out = self.run_ceph_command(cmd=cmd)
+        pool_names = [entry["name"] for entry in out.get("pools", [])]
+        pool_id = out["pools"][pool_names.index(pool)]["id"]
+
+        cmd = f" ceph pg ls {pool_id}"
+        out = self.run_ceph_command(cmd=cmd)
+        for ele in out["pg_stats"]:
+            if any(key in disallowed_states for key in ele["state"].split("+")):
+                log.error(
+                    f"PG : {ele['pgid']} is in state : {ele['state']}. "
+                    "PG expected to be active+clean"
+                )
+                return False
+            else:
+                log.info(f"PG : {ele['pgid']} is in expected state : {ele['state']}. ")
+        log.info("Completed checking PG states on all PGs of the pool. Pass")
+        return True
```

Comment:
```suggestion
        for state in disallowed_states:
            pg_list = self.get_pgid(pool_name=pool, states=state)
            if len(pg_list) != 0:
                log.error(f"PGs currently is disallowed state of {state}: {pg_list}")
                return False
            log.info(f"None of the PGs {pg_list} were in disallowed state of {state}")
        log.info("Completed checking PG states on all PGs of the pool. Pass")
        return True
```

Just suggesting an alternate logic for consideration
Note: would need to add below two lines in the get_pgid module here [#L1596](https://github.com/red-hat-storage/cephci/blob/6219e425f795e18a0f91d9dae46f2faf92f237fd/ceph/rados/core_workflows.py#L1596) 
_if not pgid_dict["pg_stats"]:
         return []_

Reply:
Nah, It's better to complete one PG at a time, rather than 1 state at a time.